### PR TITLE
added focus-lock to modals

### PR DESCRIPTION
**Pull request by nicholas-goodwin**
## Description

- Forced modals to not allow tabbing (or shift+tab) out of the modal.

## Reminders:
- [x] Be mindful of impacts of changing Major/Minor/Patch versions of each elixir app
- - [x] If updating patch version, are you sure there are no chart changes required to maintain functionality? If so, you should bump minor version instead.
  - [x] If updating Major or Minor versions , did you update the sauron chart configuration? 
- [x] If changing elixir code in an "app", did you update the relevant version
      in `mix.exs`?
- [x] If altering an API endpoint, was the relevant postman collection updated?
  - [x] If a new version of `smart_city` is being used (new fields on a struct), were the relevant postman collections updated?

### CODE REVIEW

File: apps/andi/assets/package-lock.json [hunk 1]
```diff
@@ -1,5478 +1,15 @@
 {
-  "name": "assets",
-  "lockfileVersion": 2,
   "requires": true,
-  "packages": {
```
Comment (c-m-duncan) [inline apps/andi/assets/package-lock.json:5]: Did you do this on purpose? Like a prune that had to occur?

**Review by c-m-duncan — APPROVED** [review]
Looks like a good solution to me!

**Comment by bmitchinson** [comment]
I'm open to adding a javascript library to address this if needed. Seems like the bundle size is pretty small. https://bundlephobia.com/package/alpinejs@3.10.5

Were there no other html native ways of doing this? Using tabindex seems interesting but maybe it's not an option, and x-trap is a better way to go. https://www.w3docs.com/snippets/html/how-to-prevent-tab-indexing-on-specific-elements.html

File: apps/andi/assets/package-lock.json (hunk 1 above)
Comment (nicholas-goodwin) [reply apps/andi/assets/package-lock.json:5]: no, just what happened when I built. I typically try to leave it out, but I think I had to, as I added a couple imports

**Comment by nicholas-goodwin** [comment]
tabindexing was a no-go, as I would have to go into each field in the background and make it -1 every time we open a modal, which I think will turn ugly real quick.
There are bootstrap ways to build a modal that I know of via html (well, still kinda using something besides html), but don't didn't want to bring in that whole thing just for all the stylings to go to wack.
I can give x-trap a shot, hold a minute.

**Pull request merged**